packages/plugin-lighthouse/src/lib/runner/utils.ts

```diff
@@ -0,0 +1,147 @@
+import chalk from 'chalk';
+import type { Budget, Config } from 'lighthouse';
+import log from 'lighthouse-logger';
+import desktopConfig from 'lighthouse/core/config/desktop-config.js';
+import experimentalConfig from 'lighthouse/core/config/experimental-config.js';
+import perfConfig from 'lighthouse/core/config/perf-config.js';
+import { Result } from 'lighthouse/types/lhr/audit-result';
+import path from 'node:path';
+import { AuditOutput, AuditOutputs } from '@code-pushup/models';
+import { importEsmModule, readJsonFile, ui } from '@code-pushup/utils';
+import {
+  LIGHTHOUSE_UNSUPPORTED_CLI_FLAGS,
+  UnsupportedCliFlags,
+} from '../constants';
+import type { LighthouseCliFlags } from './runner';
+
+export const unsupportedDetailTypes = new Set([
+  'opportunity',
+  'table',
+  'treemap-data',
+  'screenshot',
+  'filmstrip',
+  'debugdata',
+  'criticalrequestchain',
+]);
+
+export function toAuditOutputs(lhrAudits: Result[]): AuditOutputs {
+  // @TODO implement all details
+  const slugsWithDetailParsingErrors = lhrAudits
+    .filter(({ details }) =>
+      unsupportedDetailTypes.has(details?.type as string),
+    )
+    .map(({ details }) => details?.type);
+  ui().logger.info(
+    `Parsing details from audits and types unsupported: ${slugsWithDetailParsingErrors.join(
+      ', ',
+    )}`,
+  );
+
+  return lhrAudits.map(
+    ({
+      id: slug,
+      score,
+      numericValue: value = 0, // not every audit has a numericValue
+      details,
+      displayValue,
+    }: Result) => {
+      const auditOutput: AuditOutput = {
+        slug,
+        score: score ?? 1, // score can be null
+        value: Number.parseInt(value.toString(), 10),
+        displayValue,
+      };
+
+      if (details == null) {
+        return auditOutput;
+      }
+
+      return auditOutput;
```

Comment:
❔ Is the additional branching necessary? The returned value is identical.

Reply:
No!